### PR TITLE
Fix tsan error: change memory randomization

### PR DESCRIPTION
**Pull request by mehmetyusufoglu**
Decrease mmap randomization to fix the tsan error `FATAL: ThreadSanitizer: unexpected memory mapping`

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
This looks like the similar issue: https://github.com/google/sanitizers/issues/1716

**Comment by psychocoderHPC** [comment]
If I see it correctly the bug should be solved in llvm 17+ see https://github.com/llvm/llvm-project/commit/fb77ca05ffb4f8e666878f2f6718a9fb4d686839

**Comment by SimeonEhrig** [comment]
> If I see it correctly the bug should be solved in llvm 17+ see [llvm/llvm-project@fb77ca0](https://github.com/llvm/llvm-project/commit/fb77ca05ffb4f8e666878f2f6718a9fb4d686839)

No, it does not solve the problem. I already tested it: https://github.com/alpaka-group/alpaka/pull/2249/commits/812035ea3733a8854501deec9454b7dda3652ffe

I think, this fixes another bug.

**Comment by mehmetyusufoglu** [comment]
this is a workaround for linux OS. Tsan does not fail anymore.
`sudo sysctl vm.mmap_rnd_bits=28`

I will put a condition to set this only for tsan or only for linux.

File: script/set.sh [hunk 1]
```diff
@@ -11,4 +11,6 @@
 # -u: treat unset variables as an error
 # -v: Print shell input lines as they are read
 # -x: Print command traces before executing command
+sudo sysctl vm.mmap_rnd_bits=28
```
Comment (SimeonEhrig) [inline script/set.sh]: You was faster. I would also suggest it :-) It works.

Please guard it with:

```bash
if [ -n "$GITHUB_ACTIONS" ] && [ "$ALPAKA_CI_OS_NAME" = "Linux" ]; then
    sudo sysctl vm.mmap_rnd_bits=28
fi
```

An move it to `ci.sh`. `set.sh` is sourced many times and it's meaning is really executing the `set` command for the bash.

**Review by SimeonEhrig — CHANGES_REQUESTED** [review]

**Review by psychocoderHPC — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
@mehmetyusufoglu please update the description of this PR

**Comment by mehmetyusufoglu** [comment]
> @mehmetyusufoglu please update the description of this PR

Done, thanks.

**Review by SimeonEhrig — APPROVED** [review]

**Pull request merged**